test/e2e/nodepool_imagetype_test.go

```diff
@@ -0,0 +1,114 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/hypershift-operator/controllers/hostedcluster"
+	e2eutil "github.com/openshift/hypershift/test/e2e/util"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type NodePoolImageTypeTest struct {
+	DummyInfraSetup
+	ctx        context.Context
+	mgmtClient crclient.Client
+
+	hostedCluster       *hyperv1.HostedCluster
+	hostedClusterClient crclient.Client
+	clusterOpts         e2eutil.PlatformAgnosticOptions
+}
+
+func NewNodePoolImageTypeTest(ctx context.Context, mgmtClient crclient.Client, hostedCluster *hyperv1.HostedCluster, hcClient crclient.Client, clusterOpts e2eutil.PlatformAgnosticOptions) *NodePoolImageTypeTest {
+	return &NodePoolImageTypeTest{
+		ctx:                 ctx,
+		hostedCluster:       hostedCluster,
+		hostedClusterClient: hcClient,
+		clusterOpts:         clusterOpts,
+		mgmtClient:          mgmtClient,
+	}
+}
+
+func (it *NodePoolImageTypeTest) Setup(t *testing.T) {
+	// Skip test for non-AWS platforms since ImageType is currently AWS-specific
+	if globalOpts.Platform != hyperv1.AWSPlatform {
+		t.Skip("test is only supported for AWS platform")
+	}
+	if e2eutil.IsLessThan(e2eutil.Version419) {
```

Comment:
This check needs to be done in the setup of a `HostedClusterNodePoolTestCases`, not here.